src/js/ValidationProgress.jsx

```diff
@@ -0,0 +1,50 @@
+var React = require('react');
+var api = require('./api');
+
+class ValidationProgress extends React.Component {
+
+  constructor(props){
```

Comment:
Honestly, I like the idea. Just threw me for a minute. 😕 
I create #133. I think we should stick to a path but think ES6 is the correct one. I think we even talked about that before we started to build.